samples/encode-decode-worker/js/stream_worker.js

```diff
@@ -0,0 +1,323 @@
+  const len  = dec_aggregate.all.length;
+  const half = len >> 1;
+  const median = len % 2 === 1 ? dec_aggregate.all[len >> 1] : (dec_aggregate.all[half - 1] + dec_aggregate.all[half]) / 2;
+  return {
+     count: len,
+     min: dec_aggregate.min,
+     max: dec_aggregate.max,
+     avg: dec_aggregate.sum / len,
+     median,
+  };
+}
+
+function decqueue_update(duration) {
+   decqueue_aggregate.all.push(duration);
+   decqueue_aggregate.min = Math.min(decqueue_aggregate.min, duration);
+   decqueue_aggregate.max = Math.max(decqueue_aggregate.max, duration);
+   decqueue_aggregate.sum += duration;
+}
+
+function decqueue_report() {
+  decqueue_aggregate.all.sort();
+  const len  = decqueue_aggregate.all.length;
+  const half = len >> 1;
+  const median = len % 2 === 1 ? decqueue_aggregate.all[len >> 1] : (decqueue_aggregate.all[half - 1] + decqueue_aggregate.all[half]) / 2;
+  return {
+     count: len,
+     min: decqueue_aggregate.min,
+     max: decqueue_aggregate.max,
+     avg: decqueue_aggregate.sum / len,
+     median,
+  };
+}
+
+self.addEventListener('message', async function(e) {
+  if (stopped) return;
+  // In this demo, we expect at most two messages, one of each type.
+  let type = e.data.type;
+
+  if (type == "stop") {
+    self.postMessage({text: 'Stop message received.'});
+    if (started) pl.stop();
+    return;
+  } else if (type != "stream"){
+    self.postMessage({severity: 'fatal', text: 'Invalid message received.'});
+    return;
+  }
+  // We received a "stream" event
+  self.postMessage({text: 'Stream event received.'});
+
+  try {
+    pl = new pipeline(e.data);
+    pl.start();
+  } catch (e) {
+    self.postMessage({severity: 'fatal', text: `Pipeline creation failed: ${e.message}`})
+    return;
+  }
+}, false);
+
+class pipeline {
+
+   constructor(eventData) {
+     this.stopped = false;
+     this.inputStream = eventData.streams.input;
+     this.outputStream = eventData.streams.output;
+     this.config = eventData.config;
+   }
+
+   DecodeVideoStream(self) {
+     return new TransformStream({
+       start(controller) {
+         this.decoder = decoder = new VideoDecoder({
+           output: frame => controller.enqueue(frame),
+           error: (e) => {
+              self.postMessage({severity: 'fatal', text: `Init Decoder error: ${e.message}`});
+           }
+         });
+       },
+       transform(chunk, controller) {
+         if (this.decoder.state != "closed") {
+           if (chunk.type == "config") {
+              let config = JSON.parse(chunk.config);
+              VideoDecoder.isConfigSupported(config).then((decoderSupport) => {
+                if(decoderSupport.supported) {
+                  this.decoder.configure(decoderSupport.config);
+                  self.postMessage({text: 'Decoder successfully configured:\n' + JSON.stringify(decoderSupport.config)});
+                 // self.postMessage({text: 'Decoder state: ' + JSON.stringify(this.decoder.state)});
```

Comment:
Ditto

Reply:
This can be removed (it's not very useful even for debug).